src/sentry/static/sentry/app/views/settings/project/projectEnvironments.jsx

```diff
@@ -27,71 +21,48 @@ import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader
 import recreateRoute from 'app/utils/recreateRoute';
 import space from 'app/styles/space';
 
+const DEFAULT_EMPTY_ROUTING_NAME = 'none';
+
 const ProjectEnvironments = createReactClass({
   propTypes: {
     api: PropTypes.object,
     routes: PropTypes.array,
     params: PropTypes.object,
   },
 
-  mixins: [Reflux.listenTo(EnvironmentStore, 'onEnvironmentsChange')],
-
   getInitialState() {
-    const isHidden = this.props.location.pathname.endsWith('hidden/');
-    const environments = isHidden
-      ? EnvironmentStore.getHidden()
-      : EnvironmentStore.getActive();
-
     return {
       project: null,
-      environments,
-      isHidden,
+      environments: null,
+      isLoading: true,
     };
   },
 
   componentDidMount() {
-    if (this.state.environments === null) {
-      this.fetchData(this.state.isHidden);
-    }
+    this.fetchData();
+  },
 
-    // Fetch project details instead of using project context to guarantee we have latest project details
-    this.fetchProjectDetails();
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.location.pathname.endsWith('hidden/') !==
+      prevProps.location.pathname.endsWith('hidden/')
+    ) {
+      this.fetchData();
+    }
   },
 
-  componentWillReceiveProps(nextProps) {
+  fetchData() {
     const isHidden = this.props.location.pathname.endsWith('hidden/');
-    const environments = isHidden
-      ? EnvironmentStore.getHidden()
-      : EnvironmentStore.getActive();
-
-    this.setState(
-      {
-        isHidden,
-        environments,
-      },
-      () => {
-        if (environments === null) {
-          this.fetchData(isHidden);
-        }
-      }
-    );
-  },
 
-  refetchAll() {
-    this.fetchData(true);
-    this.fetchData(false);
-    this.fetchProjectDetails();
-  },
+    this.setState({isLoading: true});
```

Comment:
I think it might be a good idea to add a guard here so as to not trigger `componentDidUpdate()` more than is necessary:

```suggestion
    if (!this.state.isLoading) {
      this.setState({isLoading: true});
    }
```

Reply:
Yeah, I'm not sure how React handles this but sounds good to me